packages/middleware-endpoint/src/adaptors/getEndpointFromConfig.ts

```diff
@@ -0,0 +1,5 @@
+import { loadConfig } from "@smithy/node-config-provider";
+
+import { getEndpointUrlConfig } from "./getEndpointUrlConfig";
+
+export const getEndpointFromConfig = async (serviceId: string) => loadConfig(getEndpointUrlConfig(serviceId))();
```

Comment:
I think `loadConfig` needs an `await` since it returns a `Provider`?

Or if nothing needs `await`, this could be a regular function.

Reply:
The `loadConfig` does not need await. The callee `getEndpointFromInstructions` awaits it.